src/compiler/parser.ts

```diff
@@ -2143,7 +2145,11 @@ namespace Parser {
         // Don't report another error if it would just be at the same position as the last error.
         const lastError = lastOrUndefined(parseDiagnostics);
         let result: DiagnosticWithDetachedLocation | undefined;
-        if (!lastError || start !== lastError.start) {
+        if (message.category === DiagnosticCategory.Message && lastError && start === lastError.start && length === lastError.length) {

```

Comment:
This is for providing spelling suggestion in `scanCharacterClassEscape`. Otherwise we will have to add another method to the `Scanner` interface other than `setOnError`